### PR TITLE
Add Half luminance support

### PR DESCRIPTION
**Pull request by FCare**
Add Half luminance support.

It is fixing Dragon Ball Z menu selection. But still, there is other issues in the game...

The second patch is fixing Virtua Fighter CG portrait series game and not only the demo CD.

### CODE REVIEW

File: yabause/src/yglshaderes.c [hunk 1]
```diff
@@ -958,6 +958,52 @@ static YglVdp1CommonParam grow_tess = { 0 };
...
+      "out vec4 fragColor;            \n"
+      "void main()                                         \n"
+      "{                                                   \n"
+      "  vec2 addr = v_texcoord.st;                        \n"
+      "  addr.s = addr.s / (v_texcoord.q);                 \n"
+      "  addr.t = addr.t / (v_texcoord.q);                 \n"
+      "  vec4 FragColor = texture( s_texture, addr );      \n"
+      "  /*if( FragColor.a == 0.0 ) discard;*/                \n"
+      "  fragColor = FragColor;\n "
+      "  fragColor.a = 0.5;\n "
```
Comment (devmiyax) [inline yabause/src/yglshaderes.c]: Half luminance is not half transparent.

I think it should be like this. Right?
```C
fragColor.r = FragColor.r /2.0;
fragColor.g = FragColor.g /2.0;
fragColor.b = FragColor.b /2.0;
fragColor.a = FragColor.a;
```

File: yabause/src/yglshaderes.c [hunk 2]
```diff
@@ -958,6 +958,52 @@ static YglVdp1CommonParam grow_tess = { 0 };
...
+      "in vec4 v_texcoord;                            \n"
+      "uniform sampler2D s_texture;                        \n"
+      "out vec4 fragColor;            \n"
+      "void main()                                         \n"
+      "{                                                   \n"
+      "  vec2 addr = v_texcoord.st;                        \n"
+      "  addr.s = addr.s / (v_texcoord.q);                 \n"
+      "  addr.t = addr.t / (v_texcoord.q);                 \n"
+      "  vec4 FragColor = texture( s_texture, addr );      \n"
+      "  /*if( FragColor.a == 0.0 ) discard;*/                \n"
```
Comment (devmiyax) [inline yabause/src/yglshaderes.c]: Discard is needed.

**Comment by FCare** [comment]
Yes you are right I will change the commit.

**Comment by FCare** [comment]
I changed the patch. It is working far better :-) The full menu is fixed now :-)

**Comment by BenjaminSiskoo** [comment]
I've got this error message : 
3>C:\Users\Siskoo\Desktop\yabause-extui\yabause\src\yglshaderes.c(1967): error C2065: 'PG_VFP1_HALF_LUMINANCE' : identificateur non déclaré
3>C:\Users\Siskoo\Desktop\yabause-extui\yabause\src\yglshaderes.c(1970): error C2065: 'PG_VFP1_HALF_LUMINANCE' : identificateur non déclaré
3>C:\Users\Siskoo\Desktop\yabause-extui\yabause\src\yglshaderes.c(2376): error C2065: 'PG_VFP1_HALF_LUMINANCE' : identificateur non déclaré
3>C:\Users\Siskoo\Desktop\yabause-extui\yabause\src\yglshaderes.c(2382): error C2065: 'PG_VFP1_HALF_LUMINANCE' : identificateur non déclaré
3>C:\Users\Siskoo\Desktop\yabause-extui\yabause\src\yglshaderes.c(2383): error C2065: 'PG_VFP1_HALF_LUMINANCE' : identificateur non déclaré
3>C:\Users\Siskoo\Desktop\yabause-extui\yabause\src\yglshaderes.c(2384): error C2065: 'PG_VFP1_HALF_LUMINANCE' : identificateur non déclaré
3>C:\Users\Siskoo\Desktop\yabause-extui\yabause\src\yglshaderes.c(2649): warning C4090: '=' : qualificateurs 'const' différents
3>C:\Users\Siskoo\Desktop\yabause-extui\yabause\src\yglshaderes.c(2652): warning C4090: '=' : qualificateurs 'const' différents
3>C:\Users\Siskoo\Desktop\yabause-extui\yabause\src\yglshaderes.c(2646): warning C4090: 'initialisation' : qualificateurs 'const' différents

**Comment by BenjaminSiskoo** [comment]
Tested, seems ok :)

**Comment by devmiyax** [comment]
Thanks!

**Pull request merged**